langchain/src/tools/dataforseo_api_search.ts

```diff
@@ -0,0 +1,217 @@
+import { Buffer } from "buffer";
+import { getEnvironmentVariable } from "../util/env.js";
+import { Tool } from "./base.js";
+
+export class DataForSeoAPISearch extends Tool {
+  protected apiLogin: string;
+
+  protected apiPassword: string;
+
+  protected defaultParams: Record<string, any> = {
+    location_name: "United States",
+    language_code: "en",
+    depth: 10,
+    se_name: "google",
+    se_type: "organic",
+  };
+
+  protected params: Record<string, any> = {};
+
+  protected json_result_types: Array<string> | undefined = undefined;
+
+  protected json_result_fields: Array<string> | undefined = undefined;
+
+  protected top_count: number | undefined = undefined;
+
+  protected use_json_output = false;
+
+  constructor(config: Partial<DataForSeoApiConfig> = {}) {
+    super();
+    const apiLogin =
+      config.apiLogin ?? getEnvironmentVariable("DATAFORSEO_LOGIN");
+    const apiPassword =
+      config.apiPassword ?? getEnvironmentVariable("DATAFORSEO_PASSWORD");
+    const params = config.params ?? {};
+    if (!apiLogin || !apiPassword) {
+      throw new Error(
+        "DataForSEO login or password not set. You can set it as DATAFORSEO_LOGIN and DATAFORSEO_PASSWORD in your .env file, or pass it to DataForSeoAPIWrapper."
+      );
+    }
+    this.params = { ...this.defaultParams, ...params };
+    this.apiLogin = apiLogin;
+    this.apiPassword = apiPassword;
+    this.json_result_types = config.json_result_types;
+    this.json_result_fields = config.json_result_fields;
+    this.use_json_output = config.use_json_output ?? false;
+    this.top_count = config.top_count;
+  }
+
+  name = "dataforseo-api-wrapper";
+
+  description =
+    "A robust Google Search API provided by DataForSeo. This tool is handy when you need information about trending topics or current events.";
+
+  async _call(keyword: string): Promise<string> {
+    return this.use_json_output
+      ? JSON.stringify(await this.results(keyword))
+      : this.processResponse(await this.getResponseJson(keyword));
+  }
+
+  async results(keyword: string): Promise<Array<any>> {
+    const res = await this.getResponseJson(keyword);
+    return this.filterResults(res, this.json_result_types);
+  }
+
+  private prepareRequest(keyword: string): {
+    url: string;
+    headers: any;
+    data: any;
+  } {
+    if (this.apiLogin === undefined || this.apiPassword === undefined) {
+      throw new Error("api_login or api_password is not provided");
+    }
+
+    const credentials = Buffer.from(
+      `${this.apiLogin  }:${  this.apiPassword}`,
+      "utf-8"
+    ).toString("base64");
+    const headers = {
+      Authorization: `Basic ${credentials}`,
+      "Content-Type": "application/json",
+    };
+    const obj: any = { keyword: encodeURIComponent(keyword) };
```

Comment:
`any` will throw a linter warning - can we use something else or just remove the annotation?